go.mod

```diff
@@ -57,9 +57,9 @@ require (
 	gopkg.in/guregu/null.v3 v3.4.0
 	gopkg.in/segmentio/analytics-go.v3 v3.1.0
 	gotest.tools v2.2.0+incompatible
-	k8s.io/api v0.20.14
-	k8s.io/apimachinery v0.20.14
-	k8s.io/client-go v0.20.14
+	k8s.io/api v0.30.1
+	k8s.io/apimachinery v0.30.1
+	k8s.io/client-go v0.30.1
```

Comment:
ideally we could keep our client at the earliest k8s version we support. I think that would be 0.28 now

https://github.com/determined-ai/determined/pull/9347#discussion_r1597647695

Taking a step back for this change do we need to change this version or can we do
```
go get k8s.io/component-helpers@v0.20.14
```

Reply:
I can try downgrading it, I don't know if running `go mod tidy` after downgrading one of its dependencies would revert the forced downgrade though

Reply:
woo downgrading this cut out many of the extraneous changes I added in